edb/pgsql/compiler/dml.py

```diff
@@ -372,6 +375,24 @@ def fini_dml_stmt(
 
     relctx.include_rvar(ctx.rel, union_rvar, ir_stmt.subject.path_id, ctx=ctx)
 
+    for check_cte in parts.check_ctes:
```

Comment:
Add a comment?

Reply:
I don't think I understand why we need to collect the counts in the enclosing query

Reply:
We need to force the evaluation to trigger a call to `raise`, since nothing else depends on the check CTE.  Will add a comment.